bridge-ffi/src/lib.rs

```diff
@@ -0,0 +1,650 @@
+        }
+        if let Some(v) = req.nonsticky_to_sticky_poll_ratio {
+            config.nonsticky_to_sticky_poll_ratio(v);
+        }
+        if let Some(v) = req.max_concurrent_at_polls {
+            config.max_concurrent_at_polls(v as usize);
+        }
+        config.no_remote_activities(req.no_remote_activities);
+        if let Some(v) = req.sticky_queue_schedule_to_start_timeout {
+            let v: std::time::Duration = v.try_into().map_err(|_| {
+                register_worker_error(
+                    "invalid sticky queue schedule to start timeout".to_string(),
+                    false,
+                )
+            })?;
+            config.sticky_queue_schedule_to_start_timeout(v);
+        }
+        if let Some(v) = req.max_heartbeat_throttle_interval {
+            let v: std::time::Duration = v.try_into().map_err(|_| {
+                register_worker_error("invalid max heartbeat throttle interval".to_string(), false)
+            })?;
+            config.max_heartbeat_throttle_interval(v);
+        }
+        if let Some(v) = req.default_heartbeat_throttle_interval {
+            let v: std::time::Duration = v.try_into().map_err(|_| {
+                register_worker_error(
+                    "invalid default heartbeat throttle interval".to_string(),
+                    false,
+                )
+            })?;
+            config.default_heartbeat_throttle_interval(v);
+        }
+        let config = config
+            .build()
+            .map_err(|err| register_worker_error(format!("invalid request: {}", err), false))?;
+        self.core.register_worker(config).await.map_err(|err| {
+            register_worker_error(format!("{}", err),
+            matches!(err, temporal_sdk_core_api::errors::WorkerRegistrationError::WorkerAlreadyRegisteredForQueue(_)))
+        })
+    }
+
+    async fn poll_workflow_activation(
+        &self,
+        req: bridge::PollWorkflowActivationRequest,
+    ) -> Result<
+        temporal_sdk_core_protos::coresdk::workflow_activation::WorkflowActivation,
+        bridge::poll_workflow_activation_response::Error,
+    > {
+        self.core
+            .poll_workflow_activation(&req.task_queue)
+            .await
+            .map_err(|err| bridge::poll_workflow_activation_response::Error {
+                message: format!("{}", err),
+            })
+    }
+
+    async fn poll_activity_task(
+        &self,
+        req: bridge::PollActivityTaskRequest,
+    ) -> Result<
+        temporal_sdk_core_protos::coresdk::activity_task::ActivityTask,
+        bridge::poll_activity_task_response::Error,
+    > {
+        self.core
+            .poll_activity_task(&req.task_queue)
+            .await
+            .map_err(|err| bridge::poll_activity_task_response::Error {
+                message: format!("{}", err),
+            })
+    }
+
+    fn borrow_buf(&mut self) -> Vec<u8> {
+        // TODO(cretz): Implement thread-safe byte pool?
+        Vec::new()
```

Comment:
Even if we don't use a pool, makes sense to pass capacity here so the vec can be created with it immediately rather than calling `reserve` subsequently and performing another allocation

Reply:
Per the docs, it doesn't perform two allocations. From the docs:

> if you construct a Vec with capacity 0 via Vec::new, vec![], Vec::with_capacity(0), or by calling shrink_to_fit on an empty Vec, it will not allocate memory.

So `Vec::new()` + `reserve` is effective the same as `Vec::with_capacity`

Reply:
Cool, makes sense (though you'll probably need to add it as a param anyway if you use a pool later, and I think it encapsulates the better)

Reply:
Agreed, though I very likely may become convinced that a pool just isn't worth it assuming the allocator is high performing. I'll leave this borrow/return abstraction here, but remove the `TODO`s and just say that a pool impl can be added here if some microbenchmark shows it has value.